amplfi/train/models/flow.py

```diff
@@ -164,55 +197,73 @@ def test_step(self, batch, _) -> AmplfiResult:
-        # and add snr if provided
-        injection_parameters = {
-            k: float(v)
-            for k, v in zip(self.inference_params, parameters, strict=False)
-        }
-        injection_parameters["ra"] = injection_parameters["phi"]
+        log_priors = self.training_prior.log_prob(samples_dict)
+
+        # when predicting, there will be no ground truth parameters
+        injection_parameters = None
+        if parameters is not None:
+            parameters = self.scale(parameters, reverse=True)
+            parameters = parameters.cpu().numpy()[0]
+
+            # create a dictionary of injection parameters
+            # mapping from parameter string to the true
+            # injection value, and add snr if provided
+            injection_parameters = {
+                k: float(v)
+                for k, v in zip(
+                    self.inference_params, parameters, strict=False
+                )
+            }
+            injection_parameters["ra"] = injection_parameters["phi"]
 
-        if snr is not None:
-            injection_parameters["snr"] = snr[0].item()
+            if snr is not None:
+                injection_parameters["snr"] = snr[0].item()
 
         result = self.cast_as_bilby_result(
             descaled.cpu().numpy(),
             log_probs.cpu().numpy(),
+            log_priors.cpu().numpy(),
             injection_parameters,
         )
 
-        return result
+        # optionally reweight to different prior
+        # TODO: include likelihood reweighting
+        reweighted = None
+        if self.target_prior is not None:
+            self._logger.info(
+                f"Reweighting {len(result.posterior)} posterior samples"
+            )
+            reweighted = result.reweight_to_prior(self.target_prior)
+            self._logger.info(
+                f"{len(reweighted.posterior)} posterior samples "
+                "after rejection sampling"
+            )
 
-    def predict_step(self, batch, _):
-        strain, asds, _ = batch
-        context = (strain, asds)
+        return result, reweighted
 
-        samples = self.model.sample(
-            self.hparams.samples_per_event, context=context
-        )
-        log_probs = self.model.log_prob(samples, context)
-        log_probs = log_probs.squeeze(1)
-        samples = samples.squeeze(1)
-        descaled = self.scale(samples, reverse=True)
-        descaled, mask = self.filter_parameters(descaled)
-
-        log_probs = log_probs[mask]
-        result = self.cast_as_bilby_result(
-            descaled.cpu().numpy(),
-            log_probs.cpu().numpy(),
-            None,
-        )
+    def test_step(self, batch, _) -> AmplfiResult:
+        strain, asds, parameters, snr = batch
+        return self.analyze_event(strain, asds, parameters, snr)
 
-        return result
+    def predict_step(self, batch, _):
+        strain, asds, _ = batch
+        return self.analyze_event(strain, asds, None, None)
 
```

Comment:
Very late to ask this basic question: what is the difference between predict and test steps in lightning?